wallet/chainntfns_test.go

```diff
@@ -0,0 +1,300 @@
+		block := &wire.MsgBlock{
+			Header: wire.BlockHeader{
+				Timestamp: prevTimestamp.Add(blockInterval),
+			},
+		}
+
+		blockHash := block.BlockHash()
+		c.blockHashes[i] = blockHash
+		c.blocks[blockHash] = block
+	}
+
+	return c
+}
+
+// GetBestBlock returns the hash and height of the best block known to the
+// backend.
+func (c *mockChainConn) GetBestBlock() (*chainhash.Hash, int32, error) {
+	bestHash, ok := c.blockHashes[c.chainTip]
+	if !ok {
+		return nil, 0, fmt.Errorf("block with height %d not found",
+			c.chainTip)
+	}
+
+	return &bestHash, int32(c.chainTip), nil
+}
+
+// GetBlockHash returns the hash of the block with the given height.
+func (c *mockChainConn) GetBlockHash(height int64) (*chainhash.Hash, error) {
+	hash, ok := c.blockHashes[uint32(height)]
+	if !ok {
+		return nil, fmt.Errorf("block with height %d not found", height)
+	}
+
+	return &hash, nil
+}
+
+// GetBlockHeader returns the header for the block with the given hash.
+func (c *mockChainConn) GetBlockHeader(hash *chainhash.Hash) (*wire.BlockHeader, error) {
+	block, ok := c.blocks[*hash]
+	if !ok {
+		return nil, fmt.Errorf("header for block %v not found", hash)
+	}
+
+	return &block.Header, nil
+}
+
+// mockBirthdayStore is a mock in-memory implementation of the birthdayStore interface
+// that will be used for the birthday block sanity check tests.
+type mockBirthdayStore struct {
+	birthday              time.Time
+	birthdayBlock         *waddrmgr.BlockStamp
+	birthdayBlockVerified bool
+	syncedTo              waddrmgr.BlockStamp
+}
+
+var _ birthdayStore = (*mockBirthdayStore)(nil)
+
+// Birthday returns the birthday timestamp of the wallet.
+func (s *mockBirthdayStore) Birthday() time.Time {
+	return s.birthday
+}
+
+// BirthdayBlock returns the birthday block of the wallet.
+func (s *mockBirthdayStore) BirthdayBlock() (waddrmgr.BlockStamp, bool, error) {
+	if s.birthdayBlock == nil {
+		err := waddrmgr.ManagerError{
+			ErrorCode: waddrmgr.ErrBirthdayBlockNotSet,
+		}
+		return waddrmgr.BlockStamp{}, false, err
+	}
+
+	return *s.birthdayBlock, s.birthdayBlockVerified, nil
+}
+
+// SetBirthdayBlock updates the birthday block of the wallet to the given block.
+// The boolean can be used to signal whether this block should be sanity checked
+// the next time the wallet starts.
+func (s *mockBirthdayStore) SetBirthdayBlock(block waddrmgr.BlockStamp) error {
+	s.birthdayBlock = &block
+	s.birthdayBlockVerified = true
+	s.syncedTo = block
+	return nil
+}
+
+// TestBirthdaySanityCheckEmptyBirthdayBlock ensures that a sanity check is not
+// done if the birthday block does not exist in the first place.
+func TestBirthdaySanityCheckEmptyBirthdayBlock(t *testing.T) {
+	t.Parallel()
+
+	chainConn := &mockChainConn{}
+
+	// Our birthday store will reflect that we don't have a birthday block
+	// set, so we should not attempt a sanity check.
+	birthdayStore := &mockBirthdayStore{}
+
+	birthdayBlock, err := birthdaySanityCheck(chainConn, birthdayStore)
```

Comment:
Should we rather make this method return `ErrBirthdayBlockNotSet`?

Reply:
Fixed.